tests/data/test_model.py

```diff
@@ -0,0 +1,83 @@
+    model_path = Path("./tests/input_files/model_local_file.txt")
+    model = ModelData.local_file(model_path)
+    # Assert the ModelData contains the content we expect
+    content = model.get_content()
+    assert content == model_path.read_text(encoding="utf-8")
+
+
+def test_architecture():
+    """Testing that the architecture is read and added to attributes"""
+    model = ModelData.local_file(
+        "./tests/input_files/model_local_file.txt",
+        filename="model",
+        architecture="mace",
+    )
+    assert model.architecture == "mace"
+
+
+def test_download_fresh_file():
+    """Test if download works"""
+    # Ensure we do not have the file cached already
+    path_test = Path("./tests/data/tmp/mace/test_download.txt")
+    path_test.unlink(missing_ok=True)
+
+    # Construct a ModelData instance downloading a non-cached file
+    model = ModelData.download(
+        url="https://raw.githubusercontent.com/stfc/aiida-mlip/main/tests/input_files/file2.txt",
+        filename="test_download.txt",
+        cache_dir="./tests/data/tmp",
+        architecture="mace",
+    )
+
+    # Assert the ModelData contains the content we expect
+    content = model.get_content()
+    assert content == "file with content\ncontent2\n"
+    file_path = Path("./tests/data/tmp/mace/test_download.txt")
+    assert file_path.exists(), f"File {file_path} does not exist."
+    # Clear test cache
+    shutil.rmtree(Path("./tests/data/tmp/mace/"))
+
+
+def test_no_download_cached_file():
+    """Test if the caching work for avoiding double download"""
+    # Ensure file is not already downloaded
+    cached_file_path = Path("./tests/data/tmp/test_model.txt")
+    if cached_file_path.exists():
+        cached_file_path.unlink()
+
+    # Ensure we have the file cached already
+    testdir = Path("./tests/data/tmp/mace/")
+    testdir.mkdir(parents=True, exist_ok=False)
+    (testdir / "test.txt").write_text("file with content\ncontent2\n", encoding="utf-8")
+
+    # Construct a ModelData instance that should use the cached file
+    model = ModelData.download(
+        url="https://raw.githubusercontent.com/stfc/aiida-mlip/main/tests/input_files/file2.txt",
+        cache_dir="./tests/data/tmp/",
+        filename="test_model.txt",
+    )
+
+    # Assert the ModelData contains the content we expect
+    content = model.get_content()
+    assert content == "file with content\ncontent2\n"
+    file_path = Path("./tests/data/tmp/mace/test_model.txt")
+    assert file_path.exists() is False, f"File {file_path} exists but it shouldn't."
+    file_path2 = Path("./tests/data/tmp/test_model.txt")
+    assert file_path2.exists() is False, f"File {file_path2} exists but it shouldn't."
+    file_path3 = Path("./tests/data/tmp/mace/test.txt")
+    assert file_path3.exists() is True, f"File {file_path3} should exist"
+    assert model.architecture == "mace"
+
+    # Clear test cache
+    shutil.rmtree(Path("./tests/data/tmp/"))
```

Comment:
no way to do this with pathlib?

Reply:
Nope, afraid not. `for x in Path.walk(top_down=True): x.unlink(); Path.unlink()` but this is somewhat less clear.

Reply:
ok I will let it pass then...